core/src/ldml/ldml_markers.cpp

```diff
@@ -88,7 +92,9 @@ void add_back_markers(std::u32string &str, const std::u32string &src, marker_map
           break;
         } else {
           prepend_marker(str, i->marker, encoding);
+#ifndef NDEBUG
           processed_markers++;
+#endif
```

Comment:
Does l.102 become `test_assert()`?

And does it need to go in a block
```c
#ifndef NDEBUG

#endif
```

Reply:
Similar question for ldml_processor.cpp with lastCtx and its assertions

Reply:
No, because these are not unit test files.